core/trino-main/src/main/java/io/trino/execution/RenameViewTask.java

```diff
@@ -88,6 +89,12 @@ public ListenableFuture<Void> execute(
         if (metadata.isView(session, target)) {
             throw semanticException(TABLE_ALREADY_EXISTS, statement, "Target view '%s' already exists", target);
         }
+        if (metadata.getTableHandle(session, target).isPresent()) {
+            throw semanticException(GENERIC_USER_ERROR, statement, "Target view '%s' does not exist, but a table with that name exists.", target);
```

Comment:
Sorry for the confusion.

`TABLE_ALREADY_EXISTS` is fit in this specific case.